sdk/core/azure-core-http-jdk-httpclient/src/test/java/com/azure/core/http/jdk/httpclient/JdkAsyncHttpClientBuilderTests.java

```diff
@@ -339,10 +339,10 @@ void testAllowedHeadersFromBoth() {
 
     @Test
     void testAllowedHeadersFromSystemProperties() {
-        System.setProperty("jdk.httpclient.allowRestrictedHeaders", "content-length, upgrade");
+        Properties properties = new Properties();
+        properties.setProperty("jdk.httpclient.allowRestrictedHeaders", "content-length, upgrade");
```

Comment:
Is this same?

Reply:
Yes, this will perform the same testing as `new Properties()` looked up values in the System properties. This just localizes the test to be thread-safe as it no longer mutates global state